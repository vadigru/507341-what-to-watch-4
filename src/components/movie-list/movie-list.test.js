import React from "react";
import renderer from "react-test-renderer";
import MovieList from "./movie-list.jsx";

const films = [
  {
    title: `Star Wars: Episode I - The Phantom Menace`,
    posterUrl: `https://m.media-amazon.com/images/M/MV5BYTRhNjcwNWQtMGJmMi00NmQyLWE2YzItODVmMTdjNWI0ZDA2XkEyXkFqcGdeQXVyNTAyODkwOQ@@._V1_.jpg`
  },
  {
    title: `Star Wars: Episode II - Attack of the Clones`,
    posterUrl: `https://images-na.ssl-images-amazon.com/images/I/61zAkpvYLqL._AC_SY679_.jpg`
  },
  {
    title: `Star Wars: Episode III - Revenge of the Sith`,
    posterUrl: `https://upload.wikimedia.org/wikipedia/en/9/93/Star_Wars_Episode_III_Revenge_of_the_Sith_poster.jpg`
  },
  {
    title: `Star Wars: Episode IV - A New Hope`,
    posterUrl: `https://m.media-amazon.com/images/I/51c6S4kGFmL.jpg`
  },
];

it(`Should render MovieList component`, () => {
  const tree = renderer
    .create(
        <MovieList
          movies={films}
          onMovieTitleClick={() => {}}
          onMovieCardHover={() => {}}
        />)
  .toJSON();

  expect(tree).toMatchSnapshot();
});
