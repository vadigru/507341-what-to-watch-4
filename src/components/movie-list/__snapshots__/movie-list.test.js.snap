// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MovieList component 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <img
        alt="Movie Title"
        height="175"
        src="https://url.com/poster.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Movie Title
      </a>
    </h3>
  </article>
</div>
`;
